spyder/config/tests/test_user.py

```diff
@@ -47,14 +84,283 @@ def test_userconfig_get_does_not_eval_functions(userconfig):
+    ([('test', {'opt': 'éàÇÃãéèï'})], u'éàÇÃãéèï'),
+    ([('test', {'opt': True})], True),
+    ([('test', {'opt': UserConfig})], repr(UserConfig)),
+    ([('test', {'opt': 123})], 123),
+    ([('test', {'opt': 123.123})], 123.123),
+    ([('test', {'opt': [1]})], [1]),
+    ([('test', {'opt': {'key': 'val'}})], {'key': 'val'}),
+])
+def test_userconfig_get(defaults, value, tmpdir):
+    name = 'foobar'
+    path = str(tmpdir)
+    conf = UserConfig(name=name, path=path, defaults=defaults,
+                      load=False, version='1.0.0', backup=False,
+                      raw_mode=True)
+
+    assert conf.get('test', 'opt') == value
+
+
+@pytest.mark.parametrize('defaults,default,raises', [
+    # Valid values
+    ([('test2', {'opt': 'value'})], 'val', True),
+    ([('test2', {'opt': 'value'})], 'val', False),
+    ([('test', {'opt': 'value'})], 'val', False),
+])
+def test_userconfig_get2(defaults, default, raises, tmpdir):
+    name = 'foobar'
+    path = str(tmpdir)
+    conf = UserConfig(name=name, path=path, defaults=defaults,
+                      load=False, version='1.0.0', backup=False,
+                      raw_mode=True)
+
+    if raises:
+        with pytest.raises(cp.NoSectionError):
+            conf.get('test', 'opt')
+    else:
+        conf.get('test', 'opt', default)
+
+
+def test_userconfig_get_default(userconfig, tmpconfig):
+    # Not existing and no defaults
+    value = userconfig.get_default('other_section', 'other_option')
+    assert value == NoDefault
+
+    # Existing and no defaults
+    value = userconfig.get_default('section', 'option')
+    assert value == NoDefault
+
+    # Existing and defaults
+    value = tmpconfig.get_default('main', 'window/is_maximized')
+    assert value is True
+
+
+def test_userconfig_reset_to_defaults(tmpdir):
+    name = 'foobar'
+    path = str(tmpdir)
+    defaults = [('main', {'opt': False}), ('test', {'opt': False})]
+    conf = UserConfig(name=name, path=path, defaults=defaults,
+                      load=False, version='1.0.0', backup=False,
+                      raw_mode=True)
+    # Skip section, should go to default
+    assert conf.defaults == defaults
+    conf.set(None, 'opt', True)
+    assert conf.get(None, 'opt') is True
+    conf.reset_to_defaults()
+    assert conf.get(None, 'opt') is False
+
+    # Provide section, should go to sectio
+    assert conf.defaults == defaults
+    conf.set('test', 'opt', True)
+    assert conf.get('test', 'opt') is True
+    conf.reset_to_defaults()
+    assert conf.get('test', 'opt') is False
+
+
+def test_userconfig_set_as_defaults(tmpdir):
+    name = 'foobar'
+    path = str(tmpdir)
+    conf = UserConfig(name=name, path=path, defaults={},
+                      load=False, version='1.0.0', backup=False,
+                      raw_mode=True)
+
+    # TODO: Is this expected? this seems inconsistent
+    assert conf.defaults == [('main', {})]
+    conf.set_as_defaults()
+    assert conf.defaults == []
```

Comment:
@ccordoba12 this is one of the tests that I feel displays an odd behavior ht USerConfig has. I would have expected the second assert to be the same as the first one.

Reply:
Yes, I agree this inconsistency is odd. Could you fix it as part of this PR?